_drafts/release-workflows.md

```diff
@@ -0,0 +1,388 @@
+|-----------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
+| **push** to `master`                                | Release on default channel (if supported, otherwise no channel)                                                  |
+| **merge** feature branch to `master`                | Release on default channel (if supported, otherwise no channel)                                                  |
+| **push** to `next`                                  | -                                                                                                                |
+| **push** to `1.x.x` or `2.x.x` branches             | Release on default channel (if supported, otherwise no channel) if in range, report an error and error otherwise |
+| **merge** any branch to `1.x.x` or `2.x.x` branches | Release on default channel (if supported, otherwise no channel) if in range, report an error and error otherwise |
+
+Note: By default lts releases are published on the default channel as they are limited to a range of versions inferior to the versions released from `master`. Dependents will get only the expected releases by specifying a range dependency (for example `^2.0.0`).
+If a `channel` is specified, then the release will be made on this channel.
+
+##### LTS branches configuration
+
+```json
+"release": {
+  "branches": ["1.x.x", "2.x.x", "master"]
+  }
+}
+```
+
+With specific channel names:
+
+```json
+"release": {
+  "branches": [{
+    "branch": "v1",
+    "range": "1.x.x",
+    "channel": "v1"
+  }, {
+    "branch": "v2",
+    "range": "2.x.x",
+    "channel": "v2"
+  }, "master"]
+}
+```
+
+#### Pre-releases
+
+Workflow to distribute releases without incrementing the semantic version during the development of new a application version.  
+Ideal for applications distributing unstable/alpha/preview releases.
+
+Branch: `master`, `next-beta`, `next-alpha`
+
+Channel: `latest` if supported by the release target
+
+##### Pre-releases workflow
+
+| Action                                                         | Result                                                                                                                                        |
+|----------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
+| **push** to `master`                                           | Release on default channel (if supported, otherwise no channel)                                                                               |
+| **merge** `4.0.0-beta`, `5.0.0-beta` or any branch to `master` | Release on default channel (if supported, otherwise no channel)                                                                               |
```

Comment:
would it be worth enhancing the release notes when a pre-release gets merged to master and published on the `default` channel? the reason i ask is similar to the goal of https://github.com/greenkeeperio/greenkeeper/issues/566 so that a consumer not subscribed to the pre-release channel would not miss the release notes from the pre-release iterations (assuming they old read the notes of the versions they actually consumed).

i'm not sure what the best approach would be for including them, but some options that come to mind are:
* just include all changes since the last publish to `latest` in the same list. this would not work outside of pre-releases since other strategies might promote to `latest` without publishing a new version. it could also be misleading since some of the changes were not directly in the release that they would be listed with
* include two lists. one for the changes directly part of that release and another list for the changes since the last version published to `latest`. most consumers might not care about the separation
* list the changes directly part of this release, but also provide links to the release notes of the pre-release iterations since the branch from `latest`. makes it clear what is directly part of the releast, but also makes it clear that there are other changes since `latest` that they might want to look deeper into

thoughts?

Reply:
I think that is something that will be resolved in Greenkeeper, I would not amend the release notes with semantic-release

Reply:
i think that makes sense for projects that consume projects through greenkeeper, but i guess i was trying to highlight the cases where that isnt the case. 

a package maintainer that uses semantic-release to publish can't count on consumers using greenkeeper to update. plus, if semantic-release is evolving to support other package managers, greenkeeper isnt available for those.

when a consumer comes looking for release notes, its currently on them to go back through all of the releases since the version they were on previously. i just wonder in cases where there were releases made to a non-default channel if it would be more helpful to give a little more guidance for the release back to the default channel. 

i can see both approaches, but thought it was at least worth thinking through as part of defining these flows

Reply:
> its currently on them to go back through all of the releases since the version they were on previously

doesn’t that make sense though? If I’m currently on say `v1.2.3` and I upgrade directly to `v2.1.4`, then I would go and look trough the release nots for all versions since then?

Reply:
> doesn’t that make sense though?

it does. like i said, i can see both ways.

if i were manually building release notes, i just think i would treat the release to the default channel after several non-default releases. i would at want to make breaking changes that are included very clear. i dont want to get in the business of manually building them, tho. maybe what i'm thinking of is better suited for a migration guide or separate CHANGELOG...

another way to maybe consider what i'm thinking about is squashed commits. not quite the same thing since the pre-releases arent truly squashed away, but to a consumer of the default channel (assuming they are staying up to date with each version published to that channel), the publish to the default is essentially the same result as squashing the pre-releases and the current release together.

> If I’m currently on say v1.2.3 and I upgrade directly to v2.1.4, then I would go and look trough the release nots for all versions since then?

 i normally do too, but i dont think thats incredibly common unless debugging something that is noticeably broken. it might not even be obvious that there were pre-release versions pubished. going from `v1.2.3` to `v2.1.4` might make someone think twice, but if the prereleases were `v2.0.0-beta.1` and `v2.0.0-beta.2` and the default channel publish is `v2.0.0`, it could easily be assumed that the current release was the only thing included in the major version bump.